web-components/src/merch-search.js

```diff
@@ -17,16 +17,31 @@ export class MerchSearch extends LitElement {
 
     constructor() {
         super();
-        this.handleInput = () =>
+
+        this.handleInput = () => {
+            pushStateFromComponent(this, this.search.value);
+        };
+        this.handleInputAndAnalytics = () => {
             pushStateFromComponent(this, this.search.value);
+            if (this.search.value) {
+                this.dispatchEvent(
+                    new CustomEvent('search-changed', {
+                        bubbles: true,
+                        composed: true,
+                        detail: { value: this.search.value },
+                    })
+                );
+            }
+        };
         this.handleInputDebounced = debounce(this.handleInput.bind(this));
```

Comment:
do you still need this?

Reply:
Yes, we have 2 events for that search field: `input` and `change`. 
`Input` is thrown for every new letter and search is performed in the handler for this event
`Change` is thrown when the search field loses focus and then we send the analytics request.
I cannot send it from the `input` handler, it would go after every letter is typed or deleted.